src/qt/bitcoin.cpp

```diff
@@ -464,6 +465,10 @@ void BitcoinApplication::handleNonFatalException(const QString& message)
         QLatin1String("<br><br>") + GUIUtil::MakeHtmlLink(message, PACKAGE_BUGREPORT));
 }
 
+int BitcoinApplication::getExitStatus() const {
+    return node().context()->exit_status.load();
```

Comment:
In commit "gui: return EXIT_FAILURE on post-init fatal errors" (c8806e972f2be9f8a05bf11b1f6a8a2e70e01962)

This is ok for now, but if you want to future-proof the code and make it compatible with #10102, it would be better to add an `interfaces::Node::getExitStatus()` method and have GUI call `node().getExitStatus()`. The problem is that with  #10102, `node().context()` will return null when the node is running in a different process than the GUI.

I also think it might be clearer just to delete the `BitcoinApplication::getExitStatus` method entirely and just `return node().getExitStatus()` at the end of `GuiMain()`

Reply:
> I also think it might be clearer just to delete the BitcoinApplication::getExitStatus method entirely and just return node().getExitStatus() at the end of GuiMain()

That was my first idea too but `GuiMain()` is a static function with no direct access to the node interface. The interface is created inside the `BitcoinApplication::createNode` class method.

> This is ok for now, but if you want to future-proof the code and make it compatible with https://github.com/bitcoin/bitcoin/pull/10102, it would be better to add an interfaces::Node::getExitStatus() method and have GUI call node().getExitStatus().

Yeah, good. Added `getExitStatus()` to the node interface.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/27708#discussion_r1224765526

> That was my first idea too but `GuiMain()` is a static function with no direct access to the node interface.

I think it can just do `return app.node().getExitStatus();`. The `app.node()` method is public and already used in `GuiMain()`

Reply:
Done, added